### PR TITLE
refactor: account compression append & unify out Merkle tree and root indices

### PR DESCRIPTION
**Pull request by ananas-block**
Note:
- this pr contains breaking changes for Photon, interop tests are disabled
this pr builds on https://github.com/Lightprotocol/light-protocol/pull/745

Changes:
- refactor accountcompression append
- unify out Merkle tree indices and output compressed account into a new struct
- unify root indices and input compressed accounts with Merkle context into a single struct

### CODE REVIEW

File: js/compressed-token/src/instructions/pack-compressed-token-accounts.ts [hunk 1]
```diff
@@ -20,6 +20,11 @@ export type PackCompressedTokenAccountsParams = {
     outputStateTrees?: PublicKey[] | PublicKey;
     /** Optional remaining accounts to append to */
     remainingAccounts?: PublicKey[];
+    /** Root indices that are used onchain to fetch the correct root
+     *  from the state Merkle tree account for validity proof verification.
+     */
```
Comment (SwenSchaeferjohann) [inline js/compressed-token/src/instructions/pack-compressed-token-accounts.ts:25]:
```suggestion
    /**
     *  Root indices that are used on-chain to fetch the correct root
     *  from the state Merkle tree account for validity proof verification.
     */
```

and is fetched from the Merkle tree account directly or from the root history array?

File: js/stateless.js/src/programs/system.ts [hunk 2]
```diff
@@ -271,36 +287,35 @@ export class LightSystemProgram {
...
             newAddressParams: [],
             inputCompressedAccountsWithMerkleContext:
                 packedInputCompressedAccounts,
             outputCompressedAccounts,
-            outputStateMerkleTreeAccountIndices: Buffer.from(
-                outputStateMerkleTreeIndices,
-            ),
+            // outputStateMerkleTreeAccountIndices: Buffer.from(
+            //     outputStateMerkleTreeIndices,
+            // ),
```
Comment (SwenSchaeferjohann) [inline js/stateless.js/src/programs/system.ts:318]: can remove?

File: programs/compressed-token/src/process_mint.rs [hunk 3]
```diff
@@ -295,21 +306,21 @@ pub fn cpi_execute_compressed_transaction_mint_to<'info>(
...
+    inputs.extend_from_slice(&[0u8; 8]);
     inputs.extend_from_slice(&[(len as u8), 0, 0, 0]);
     // output_compressed_accounts
     for compressed_account in output_compressed_accounts.iter() {
         compressed_account.serialize(inputs).unwrap();
     }
-    inputs.extend_from_slice(&[(len as u8), 0, 0, 0]);
-    inputs.extend_from_slice(&vec![0u8; len]);
+    // inputs.extend_from_slice(&[(len as u8), 0, 0, 0]);
+    // inputs.extend_from_slice(&vec![0u8; len]);
```
Comment (SwenSchaeferjohann) [inline programs/compressed-token/src/process_mint.rs:323]: leftover?

File: programs/system/src/sdk/invoke.rs [hunk 4]
```diff
@@ -279,10 +296,10 @@ mod test {
...
             2
         );
-        assert_eq!(
-            deserialized_instruction_data.input_root_indices,
-            input_root_indices
-        );
+        // assert_eq!(
+        //     deserialized_instruction_data.input_root_indices,
+        //     input_root_indices
+        // );
```
Comment (SwenSchaeferjohann) [inline programs/system/src/sdk/invoke.rs:302]: leftover?

**Review by SwenSchaeferjohann — APPROVED** [review]
lgtm. left comments for the rest

File: js/compressed-token/src/instructions/pack-compressed-token-accounts.ts (hunk 1 above)
Comment (ananas-block) [reply js/compressed-token/src/instructions/pack-compressed-token-accounts.ts:25]: from the root history array which is in the Merkle tree account

**Pull request merged**